src/MusicStore/Components/AnnouncementComponent.cs

```diff
@@ -10,18 +10,23 @@ namespace MusicStore.Components
     [ViewComponent(Name = "Announcement")]
     public class AnnouncementComponent : ViewComponent
     {
-        private readonly MusicStoreContext _dbContext;
-        private readonly IMemoryCache _cache;
+        [Activate]
+        public MusicStoreContext DbContext
+        {
+            get;
+            set;
+        }
 
-        public AnnouncementComponent(MusicStoreContext dbContext, IMemoryCache memoryCache)
+        [Activate]
+        public IMemoryCache MemoryCache
```

Comment:
For example, It has two actions. One of them doesn't need the memory cache to run. It it is a constructor parameter, users need to mock it even fo the action that doesn't require the cache.